File: JPAGoOrbital/src/main/java/com/skilldistillery/goOrbital/entities/Launchport.java
package com.skilldistillery.goOrbital.entities;

import java.util.List;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;

import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
public class Launchport {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;

	private String name;

	private double latitude;

	private double longitude;
	
	@JsonIgnore
	@OneToMany(mappedBy = "launchport")
	private List<Trip> trips;

	public Launchport() {
	}

	public Launchport(int id, String name, double latitude, double longitude) {
		this.id = id;
		this.name = name;
		this.latitude = latitude;
		this.longitude = longitude;
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public double getLatitude() {
		return latitude;
	}

	public void setLatitude(double latitude) {
		this.latitude = latitude;
	}

	public double getLongitude() {
		return longitude;
	}

	public List<Trip> getTrips() {
		return trips;
	}

	public void setTrips(List<Trip> trips) {
		this.trips = trips;
	}

	public void setLongitude(double longitude) {
		this.longitude = longitude;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + id;
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Launchport other = (Launchport) obj;
		if (id != other.id)
			return false;
		return true;
	}

	@Override
	public String toString() {
		return "Launchport [id=" + id + ", name=" + name + ", latitude=" + latitude + ", longitude=" + longitude + "]";
	}

}

File: ngGoOrbital/src/app/models/launchport.ts
import { Trip } from './trip';

//import { Launchport } from './launchport';
// import { Deserializable } from './../interfaces/deserializable';
// export class Launchport implements Deserializable {
//   id: number;
//   name: string;
//   latitude: number;
//   longitude: number;
//   deserialize(input: any): this {
//     Object.assign(this, input);
//     return this;
//   }
// }


export class Launchport {
  id: number;
  name: string;
  latitude: number;
  longitude: number;
  trips: Trip[];

  constructor({ id, name, latitude, longitude, trips}: {id?:number; name?: string; latitude?: number; longitude?: number; trips?: Trip[]}={})

  {
    this.id = id;
    this.name = name;
    this.latitude = latitude;
    this.longitude = longitude;
    this.trips = trips;
  }

}
